Mathlib/Order/Bounds/Basic.lean

```diff
@@ -1130,6 +1130,52 @@ theorem map_isLeast : IsLeast s a → IsGreatest (f '' s) (f a) :=
 
 end Antitone
 
+section StrictMono
+
+variable [LinearOrder α] [Preorder β] {f : α → β} (hf : StrictMono f) {a : α} {s : Set α}
+
+include hf
+
+theorem mem_upperBounds_image_iff : f a ∈ upperBounds (f '' s) ↔ a ∈ upperBounds s :=
+  ⟨fun Ha _ hx => hf.le_iff_le.mp (Ha (mem_image_of_mem f hx)), hf.monotone.mem_upperBounds_image⟩
+
+theorem mem_lowerBounds_image_iff : f a ∈ lowerBounds (f '' s) ↔ a ∈ lowerBounds s:=
+  ⟨fun Ha _ hx => hf.le_iff_le.mp (Ha (mem_image_of_mem f hx)), hf.monotone.mem_lowerBounds_image⟩
+
+theorem map_isLeast_iff_isLeast : IsLeast (f '' s) (f a) ↔ IsLeast s a :=
+  ⟨fun Ha =>  ⟨(mem_image_iff_mem_of_injective _ _ hf.injective).mp Ha.left,
+    (mem_lowerBounds_image_iff hf).mp Ha.right⟩, hf.monotone.map_isLeast⟩
+
+theorem map_isGreatest_iff_isGreatest : IsGreatest (f '' s) (f a) ↔ IsGreatest s a :=
+  ⟨fun Ha => ⟨(mem_image_iff_mem_of_injective _ _ hf.injective).mp Ha.left,
+    (mem_upperBounds_image_iff hf).mp Ha.right⟩, hf.monotone.map_isGreatest⟩
+
+end StrictMono
+
+section StrictAnti
+
+variable [LinearOrder α] [Preorder β] {f : α → β} (hf : StrictAnti f) {a : α} {s : Set α}
+
+include hf
```

Comment:
I've heard its good (for readability) to include explicit args in the statements themselves. So maybe include `hf` manually in each of the statements? I am not fuzzed about it though.